controllers/app.controllers.js

```diff
@@ -7,8 +11,16 @@ exports.getApi = (req, res, next) => {
 };
 
 exports.getTopics = (req, res, next) => {
-  getTopicsFromDatabase().then(topics => {
-    console.log(topics)
-    res.status(200).send({topics})
-  })
-}
\ No newline at end of file
+  getTopicsFromDatabase().then((topics) => {
+    res.status(200).send({ topics });
+  }).catch(next);
+};
+
+exports.getArticleFromId = (req, res, next) => {
+  const { params } = req;
+  getArticlesFromDatabase(params.article_id).then((article) => {
+    if (article.length === 0)
+      return Promise.reject({ msg: "bad request", status: 400 });
+    res.status(200).send({ article: article[0] });
```

Comment:
Error handling shouldnt be done in the controller like this, this should be dealt with in the model